R/keyword_search.r

```diff
@@ -76,26 +78,46 @@ keyword_search <- function(x, keyword, path = FALSE, split_pdf = FALSE,
                            remove_hyphen = TRUE, token_results = TRUE,
                            heading_search = FALSE, heading_args = NULL,
                            convert_sentence = TRUE, 
-                           remove_equations = TRUE,
-                           split_pattern = "\\p{WHITE_SPACE}{3,}", ...) {
+                           remove_equations = FALSE,
+                           split_pattern = "\\p{WHITE_SPACE}{3,}",
+                           ocr_pdf_fun = function(...) {return("")},
+                           ocr_pdf_fun_params = nil,
+                           translate_fun = function(x,...) {return(x)},
+                           translate_fun_params = nil,
+                           translation_target_language="eng",
+                           ...) {
+  print(x)
   if(path) {
-    x <- pdftools::pdf_text(x)
+    path <- x
+    x  <- suppressMessages(pdftools::pdf_text(x))
+    output_dir_name <- paste0(dirname(path),"/outputs/")

```

Comment:
My colleges found it important, that the tools is transparent on its three "magic tasks":
- text extraction from text
- OCR
- translation

So I write the results of this into up to three text files in an "output" folder.